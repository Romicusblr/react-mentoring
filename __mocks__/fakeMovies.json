{
  "data": [
    {
      "id": 58574,
      "title": "Sherlock Holmes: A Game of Shadows",
      "tagline": "The game is afoot.",
      "vote_average": 7,
      "vote_count": 4823,
      "release_date": "2011-11-22",
      "poster_path": "https://image.tmdb.org/t/p/w500/7OFqsVuS6DlVM10GUD72vCGpQm9.jpg",
      "overview": "There is a new criminal mastermind at large (Professor Moriarty) and not only is he Holmesâ€™ intellectual equal, but his capacity for evil and lack of conscience may give him an advantage over the  detective.",
      "budget": 125000000,
      "revenue": 334615000,
      "genres": [
        "Adventure",
        "Action",
        "Crime",
        "Mystery"
      ],
      "runtime": 129
    },
    {
      "id": 185,
      "title": "A Clockwork Orange",
      "tagline": "Being the adventures of a young man whose principal interests are rape, ultra-violence and Beethoven.",
      "vote_average": 8.1,
      "vote_count": 4553,
      "release_date": "1971-12-18",
      "poster_path": "https://image.tmdb.org/t/p/w500/kgAi87gyx6b4oGJYSC36tVkJyJu.jpg",
      "overview": "Demonic gang-leader Alex goes on the spree of rape, mugging and murder with his pack of \"droogs\". But he's a boy who also likes Beethoven's Ninth and a bit of \"the old in-out, in-out\". He later finds himself at the mercy of the state and its brainwashing experiment designed to take violence off the streets.",
      "budget": 2200000,
      "revenue": 26589000,
      "genres": [
        "Science Fiction",
        "Drama"
      ],
      "runtime": 136
    },
    {
      "id": 10528,
      "title": "Sherlock Holmes",
      "tagline": "Nothing escapes him.",
      "vote_average": 7,
      "vote_count": 6907,
      "release_date": "2009-12-23",
      "poster_path": "https://image.tmdb.org/t/p/w500/22ngurXbLqab7Sko6aTSdwOCe5W.jpg",
      "overview": "Eccentric consulting detective, Sherlock Holmes and Doctor John Watson battle to bring down a new nemesis and unravel a deadly plot that could destroy England.",
      "budget": 90000000,
      "revenue": 524028679,
      "genres": [
        "Action",
        "Adventure",
        "Crime",
        "Mystery"
      ],
      "runtime": 128
    }
  ],
  "total": 3,
  "offset": 0,
  "limit": 10
}
