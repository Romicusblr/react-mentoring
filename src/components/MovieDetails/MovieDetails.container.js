import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { fetchMovie } from '../../actions';
import MovieDetails from './MovieDetails';

const mapStateToProps = ({ currMovie }) => ({
  data: currMovie.data,
  loading: currMovie.loading,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchMovie,
}, dispatch);

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MovieDetails));
