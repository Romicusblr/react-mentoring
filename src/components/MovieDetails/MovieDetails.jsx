/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import LabelRaised from '../LabelRaised';
import Rating from '../Rating';
import List from '../List';

import style from './MovieDetails.module.css';

function MovieDetails({ movie, className, ...other }) {
  if (!movie) return null;

  const {
    title, genres, release_date, vote_average,
    poster_path, overview, runtime,
  } = movie;

  return (
    <figure
      className={classNames(style.movieDetails, className)}
      {...other}
    >
      <img src={poster_path} alt={title} />
      <figcaption>
        <h1 className={style.title}>
          <span>{title}</span>
          <Rating>{vote_average.toFixed(1)}</Rating>
        </h1>
        <p className={style.label}>
          <LabelRaised>{new Date(release_date).getFullYear()}</LabelRaised>
          <LabelRaised>{`${runtime} min`}</LabelRaised>
        </p>
        <List items={genres} limit={2} delimeter=" &amp; " />
        <p>{overview}</p>
      </figcaption>
    </figure>
  );
}

MovieDetails.propTypes = {
  movie: PropTypes.objectOf(PropTypes.any),
  className: PropTypes.string,
};

MovieDetails.defaultProps = {
  movie: null,
  className: '',
};

export default MovieDetails;
