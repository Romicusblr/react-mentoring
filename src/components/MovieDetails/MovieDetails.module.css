.movieDetails {
  margin: 0;
  display: flex;
  &>img {
    width: 17.5rem;
  }
  &>figcaption {
    margin-left: 3.75rem;
  }
}

.title {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  &>span:first-child {
    margin-right: .5rem;
  }
}

.rating {
  margin-left: .5rem;
}

.label > * {
  margin-right: 1rem;
}
