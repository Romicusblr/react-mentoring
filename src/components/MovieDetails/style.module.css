.movieDetails {
  margin: 0;
  display: flex;
  &>img {
    width: 17.5rem;
  }
  &>figcaption {
    margin-left: 3.75rem;
    &>h2 {
      display: flex;
      justify-content: flex-start;
      align-items: center;
      &>p {
        margin: 0;
        margin-right: .5rem;
      }
    }
  }

}
