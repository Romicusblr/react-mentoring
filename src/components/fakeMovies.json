{
  "data": [
    {
      "id": 58574,
      "title": "Sherlock Holmes: A Game of Shadows",
      "tagline": "The game is afoot.",
      "vote_average": 7,
      "vote_count": 4823,
      "release_date": "2011-11-22",
      "poster_path": "https://image.tmdb.org/t/p/w500/7OFqsVuS6DlVM10GUD72vCGpQm9.jpg",
      "overview": "There is a new criminal mastermind at large (Professor Moriarty) and not only is he Holmes’ intellectual equal, but his capacity for evil and lack of conscience may give him an advantage over the  detective.",
      "budget": 125000000,
      "revenue": 334615000,
      "genres": [
        "Adventure",
        "Action",
        "Crime",
        "Mystery"
      ],
      "runtime": 129
    },
    {
      "id": 185,
      "title": "A Clockwork Orange",
      "tagline": "Being the adventures of a young man whose principal interests are rape, ultra-violence and Beethoven.",
      "vote_average": 8.1,
      "vote_count": 4553,
      "release_date": "1971-12-18",
      "poster_path": "https://image.tmdb.org/t/p/w500/kgAi87gyx6b4oGJYSC36tVkJyJu.jpg",
      "overview": "Demonic gang-leader Alex goes on the spree of rape, mugging and murder with his pack of \"droogs\". But he's a boy who also likes Beethoven's Ninth and a bit of \"the old in-out, in-out\". He later finds himself at the mercy of the state and its brainwashing experiment designed to take violence off the streets.",
      "budget": 2200000,
      "revenue": 26589000,
      "genres": [
        "Science Fiction",
        "Drama"
      ],
      "runtime": 136
    },
    {
      "id": 10528,
      "title": "Sherlock Holmes",
      "tagline": "Nothing escapes him.",
      "vote_average": 7,
      "vote_count": 6907,
      "release_date": "2009-12-23",
      "poster_path": "https://image.tmdb.org/t/p/w500/22ngurXbLqab7Sko6aTSdwOCe5W.jpg",
      "overview": "Eccentric consulting detective, Sherlock Holmes and Doctor John Watson battle to bring down a new nemesis and unravel a deadly plot that could destroy England.",
      "budget": 90000000,
      "revenue": 524028679,
      "genres": [
        "Action",
        "Adventure",
        "Crime",
        "Mystery"
      ],
      "runtime": 128
    },
    {
      "id": 12162,
      "title": "The Hurt Locker",
      "tagline": "You'll know when you're in it.",
      "vote_average": 7.2,
      "vote_count": 2299,
      "release_date": "2008-10-10",
      "poster_path": "https://image.tmdb.org/t/p/w500/uduCNFTuEcbvhvK8rrvNdf3ahzW.jpg",
      "overview": "Forced to play a dangerous game of cat-and-mouse in the chaos of war, an elite Army bomb squad unit must come together in a city where everyone is a potential enemy and every object could be a deadly bomb.",
      "budget": 15000000,
      "revenue": 42000000,
      "genres": [
        "Drama",
        "Thriller",
        "War"
      ],
      "runtime": 131
    },
    {
      "id": 437557,
      "title": "Blockers",
      "tagline": "Teens out to have fun. Parents out to stop it.",
      "vote_average": 5.8,
      "vote_count": 13,
      "release_date": "2018-03-30",
      "poster_path": "https://image.tmdb.org/t/p/w500/9qKMYfiJTcUQbuBl82yxrmLSCX5.jpg",
      "overview": "When three parents discover their daughters’ pact to lose their virginity at prom, they launch a covert one-night operation to stop the teens from sealing the deal.",
      "budget": 0,
      "revenue": 0,
      "genres": [
        "Comedy"
      ],
      "runtime": 102
    },
    {
      "id": 291276,
      "title": "Unlocked",
      "tagline": "",
      "vote_average": 6.3,
      "vote_count": 276,
      "release_date": "2017-04-27",
      "poster_path": "https://image.tmdb.org/t/p/w500/fBDLdEraxCBKGmOPb6pJPYiDQh1.jpg",
      "overview": "After failing to apprehend the terrorist behind a Paris attack that claimed dozens of lives, CIA agent Alice Racine is forced to live in London as a caseworker. Her mentor unexpectedly calls her back into action when the CIA discovers that another attack is imminent. Alice soon learns that the classified information she's uncovered has been compromised. Running for her life, she turns to a former soldier to help her prevent a lethal biological attack on the citizens of London.",
      "budget": 0,
      "revenue": 0,
      "genres": [
        "Action",
        "Thriller"
      ],
      "runtime": 98
    },
    {
      "id": 370567,
      "title": "Sherlock Gnomes",
      "tagline": "",
      "vote_average": 4.4,
      "vote_count": 12,
      "release_date": "2018-03-15",
      "poster_path": "https://image.tmdb.org/t/p/w500/xHdf2wRgCSp9MrZRryikiZIH6jB.jpg",
      "overview": "Garden gnomes, Gnomeo & Juliet, recruit renown detective, Sherlock Gnomes, to investigate the mysterious disappearance of other garden ornaments.",
      "budget": 59000000,
      "revenue": 0,
      "genres": [
        "Romance",
        "Animation",
        "Comedy",
        "Family",
        "Fantasy",
        "Mystery"
      ],
      "runtime": 86
    },
    {
      "id": 210479,
      "title": "Locke",
      "tagline": "No turning back",
      "vote_average": 6.9,
      "vote_count": 985,
      "release_date": "2014-04-10",
      "poster_path": "https://image.tmdb.org/t/p/w500/2Prgt4JXhwiwkgotknBO4scXR8S.jpg",
      "overview": "Ivan Locke has worked hard to craft a good life for himself. Tonight, that life will collapse around him. On the eve of the biggest challenge of his career, Ivan receives a phone call that sets in motion a series of events that will unravel his family, job, and soul.",
      "budget": 2000000,
      "revenue": 4635300,
      "genres": [
        "Drama",
        "Thriller"
      ],
      "runtime": 85
    },
    {
      "id": 100,
      "title": "Lock, Stock and Two Smoking Barrels",
      "tagline": "A Disgrace to Criminals Everywhere.",
      "vote_average": 7.5,
      "vote_count": 1981,
      "release_date": "1998-03-05",
      "poster_path": "https://image.tmdb.org/t/p/w500/qV7QaSf7f7yC2lc985zfyOJIAIN.jpg",
      "overview": "A card shark and his unwillingly-enlisted friends need to make a lot of cash quick after losing a sketchy poker match. To do this they decide to pull a heist on a small-time gang who happen to be operating out of the flat next door.",
      "budget": 1350000,
      "revenue": 28356188,
      "genres": [
        "Comedy",
        "Crime"
      ],
      "runtime": 105
    },
    {
      "id": 2207,
      "title": "16 Blocks",
      "tagline": "For a New York cop and his witness, the distance between life and death just got very short.",
      "vote_average": 6.2,
      "vote_count": 837,
      "release_date": "2006-03-01",
      "poster_path": "https://image.tmdb.org/t/p/w500/wpKBLDjNRddmH6N32ni4bASogt8.jpg",
      "overview": "An aging cop is assigned the ordinary task of escorting a fast-talking witness from police custody to a courthouse, but they find themselves running the gauntlet as other forces try to prevent them from getting there.",
      "budget": 55000000,
      "revenue": 65664721,
      "genres": [
        "Action",
        "Adventure",
        "Crime",
        "Thriller"
      ],
      "runtime": 105
    }
  ],
  "total": 13,
  "offset": 0,
  "limit": 10
}
