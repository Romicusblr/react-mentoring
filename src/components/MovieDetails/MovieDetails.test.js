import MovieDetails from '.';
import fakeMovies from '../../../__mocks__/fakeMovies';

const { data: [fakeMovie] } = fakeMovies;

describe('MovieDetails', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<MovieDetails movie={fakeMovie} />);
    expect(snap).toMatchSnapshot();
  });
});
