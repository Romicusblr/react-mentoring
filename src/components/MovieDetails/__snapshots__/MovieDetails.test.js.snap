// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match to snapshot with data 1`] = `
<figure
  className="movieDetails"
>
  <img
    alt="Sherlock Holmes: A Game of Shadows"
    src="https://image.tmdb.org/t/p/w500/7OFqsVuS6DlVM10GUD72vCGpQm9.jpg"
  />
  <figcaption>
    <h1
      className="title"
    >
      <span>
        Sherlock Holmes: A Game of Shadows
      </span>
      <Rating
        className=""
      >
        7.0
      </Rating>
    </h1>
    <p
      className="label"
    >
      <LabelRaised
        className=""
      >
        2011
      </LabelRaised>
      <LabelRaised
        className=""
      >
        129 min
      </LabelRaised>
    </p>
    <List
      className=""
      delimeter=" & "
      items={
        Array [
          "Adventure",
          "Action",
          "Crime",
          "Mystery",
        ]
      }
      limit={2}
    />
    <p>
      There is a new criminal mastermind at large (Professor Moriarty) and not only is he Holmesâ€™ intellectual equal, but his capacity for evil and lack of conscience may give him an advantage over the  detective.
    </p>
  </figcaption>
</figure>
`;

exports[`MovieDetails should match to snapshot without data 1`] = `""`;
